cylc/flow/cfgspec/globalcfg.py

```diff
@@ -1198,6 +1198,9 @@ def default_for(
                    The workflow polls for task status (no task messaging)
                 ssh
                    Use non-interactive ssh for task communications
+                http
+                   Via the running Hub proxy and/or UI-Server (requires UI
+                   Server installation)
```

Comment:
How does cylc-flow know which port to contact the hub on when it doesn't share a filesystem with the hub? Do we need to configure this?

Reply:
I think the user might need to copy/replicate the `~/.cylc/uiserver/api_info.json` file (if we don't figure out a way to automate it), because the server in question might not have ssh access..

Reply:
ATM clients are generally short lived (with the exception of the `cylc tui` client) so this isn't a biggie.

I guess reload this file if we get a comms failure and retry (once) if the file contents has changed.